exporter/exporterparser/exparser.go

```diff
@@ -43,28 +42,19 @@ import (
 // get those exporters converted to directly receive
 // OpenCensus Proto TraceData.
 func OcProtoSpansToOCSpanDataInstrumented(ctx context.Context, exporterName string, te trace.Exporter, td data.TraceData) (aerr error) {
-	ctx, span := trace.StartSpan(ctx,
-		"opencensus.service.exporter."+exporterName+".ExportTrace",
-		trace.WithSampler(trace.NeverSample()))
+	var nGoodSpans int64
 
-	span.Annotate([]trace.Attribute{
-		trace.Int64Attribute("n_spans", int64(len(td.Spans))),
-	}, "")
+	observabilityRecorder := internal.NewExporterEventRecorder(exporterName)
 
-	defer func() {
-		if aerr != nil {
-			span.SetStatus(trace.Status{Code: trace.StatusCodeInternal, Message: aerr.Error()})
-		}
-		span.End()
-	}()
+	ctx = observabilityRecorder.Start(ctx, td.Node)
+	defer observabilityRecorder.End(ctx, nGoodSpans, aerr)
```

Comment:
nGoodSpans is going to be always zero when deferred function is called, either it becomes a capture or is passed as a reference.